petastorm/etl/dataset_metadata.py

```diff
@@ -298,6 +306,26 @@ def _split_row_groups(dataset):
     return split_pieces
 
 
+def _split_row_groups_from_footers(dataset):
+    """Split the row groups by reading the footers of the parquet pieces"""
+    thread_pool = futures.ThreadPoolExecutor()
+
+    def split_piece(piece):
+        metadata = piece.get_metadata(dataset.fs.open)
+        return [pq.ParquetDatasetPiece(piece.path,
+                                       row_group,
+                                       piece.partition_keys) for row_group in range(metadata.num_row_groups)]
+
+    futures_list = [thread_pool.submit(split_piece, piece) for piece in dataset.pieces]
+
+    if futures_list:
```

Comment:
Is there a reason for an explicit wait? Is it always ok to access futures after the pool is shutdown (even with the wait)? Would expect something simpler here:
```python
result =  [item for f in futures_list for item in f.result()]
thread_pool.shutdown()
return result
```

Reply:
Ah true, I was thinking that we need to wait for all the futures to complete (which is true) though the explicit wait is indeed not necessary